sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -869,20 +869,123 @@ property set.
 
 #### Device configuration file
 
-The configuration file uses a simple YAML format where each top-level key is
-a name of a device architecture. There are sub-keys under each device for
-the supported aspects and sub-group sizes.  For example:
+The "device configuration file" is a [YAML](https://yaml.org) format file which
+stores information about all the AOT targets supported by the DPC++ compiler.
+The top-level keys (nodes) can be of 2 kinds: targets and property collections.
+- Targets are those strings which can appear in the `-fsycl-targets` list.
+- Property collections are basically named groups of properties to conveniently
+  refer to from other targets and property collections.
+
+> ##### Target name specification formats
+> Targets within can be specified in three forms: long, short and dynamic.
+> - Long form has the format `<vendor>`\_`<device type>`\_`<ID>`
+> - Short form has the format `<ID>`
+> - Dynamic form has the format `<vendor>`\_`<device type>`\_`*`. Dynamic
+>   targets are used for dynamically generated platforms, such as some Intel
+>   FPGA ones. More on this below.
+>
+> Some targets have only long form, some - both long and short. There is an aspect
+> with the same spelling for each long and short form target. There is no aspect
+> for dynamic targets.
+
+
+A target or a property collection may "extend" one ore more other target or
+property collection. Thus the nodes form an asyclic directed graph, with graph
+entry nodes being targets, and inner nodes being other targets or property
+collections. A node in the graph, beside of properties listed immediately in the
+node, also possesses all the properties combined from of all its descendants.
+A property can be "combinable" - such as `aspects` - or non-combinable - such as
+`aot-toolchain`. It is an error if there are two or more not matching
+non-combinable properties possessed by a node. (**TODO** add some 'override' prefix to
+ignore base nodes' values). This is a convenience feature to avoid duplication of
+properties, such as symbolic aspects enumerations, in multiple similar targets.
+
+The "extends" relation is specifed via the `extends` key. For example:
+```
+intel_gpu:
+  aot-toolchain: intel-gpu
+  ...
+intel_gpu_12_2_1:
+  TARGET
+  ...
+intel_gpu_dg2_g10:
+  TARGET
+  extends: intel_gpu_12_2_1, intel_gpu
+  ...
+intel_gpu_dg2_g11:
+  TARGET
+  extends: intel_gpu_12_2_1, intel_gpu
+  ...
+```
+
+Each target must immediately contain the `TARGET` sub-key w/o a value.
+(**TODO**: it can specify AOT compiler argument needed to compile for this
+target).
```

Comment:
OK to remove TODO. But why TARGET is not a good way of doing this?

Reply:
Two reasons:

1. It seems unrelated to the "target" attribute.  A more descriptive key would be something like "aot-command-line".
2. We may want to put the "aot-command-line" key in a collection, so it can be shared by multiple targets.

Reply:
I think this is a matter of taste actually. I think sharing will be rare in practice. But let's remove for now anyway.